components/ui/pathogen-map/pathogen-map.tsx

```diff
@@ -0,0 +1,102 @@
+import { MapResources } from "@/app/pathogen/arbovirus/dashboard/(map)/map-config";
+import { getEsriVectorSourceStyle } from "@/utils/mapping-util";
+import mapboxgl from "mapbox-gl";
+import { useState, useEffect, useMemo } from "react";
+import {
+  Map,
+  Source,
+  Layer,
+  Popup,
+  useMap,
+  MapRef,
+  NavigationControl,
+} from "react-map-gl";
+import {
+  PathogenMapCursor,
+  usePathogenMapMouse,
+} from "./use-pathogen-map-mouse-events";
+import {
+  PathogenMapPopup,
+  PopupContentGenerator,
+  PopupInfo,
+} from "./pathogen-map-popup";
+import { PathogenMapLayer, PathogenMapLayerInfo } from "./pathogen-map-layer";
+
+export interface PathogenDataPointPropertiesBase {
+  id: string;
+  latitude: number | undefined;
+  longitude: number | undefined;
```

Comment:
Idea behind this type is that if you have data associated to a point on the map you need it to have _at least_ these properties. Keeping it to a minimum so that the arbo stuff and sero stuff can share this component. That's the same reason for all of the generic type stuff. I know you could technically remove the generic type stuff and make this work but I think that's because of how many `any`s are floating around in the code so I'd rather keep the generic around so that when those `any`s get assigned to the correct type my stuff will have very specific typing.